modules/phase_field/src/userobjects/DiscreteNucleationInserter.C

```diff
@@ -77,15 +93,22 @@ DiscreteNucleationInserter::execute()
 
     const Real random = getRandomReal();
 
-    // We check the random number against the inverse of the zero probability.
-    // for performance reasons we do a quick check against the linearized form of
-    // that probability, which is always strictly larger than the actual probability.
-    // The expression below should short circuit and the expensive exponential
-    // should rarely get evaluated
-    if (random < rate * _fe_problem.dt() && random < (1.0 - std::exp(-rate * _fe_problem.dt())))
+    // branch the operation for using Poisson statistics (with time) or
+    // time-independent probability (e.g., recrystallization fraction)
+    if (!_poisson_stats)
     {
-      _local_nucleus_list.push_back(NucleusLocation(_fe_problem.time() + _hold_time, _q_point[qp]));
-      _changes_made.first++;
+      if (getRandomReal() < rate)
```

Comment:
we already have `random` here

Reply:
I don't like this. It seems unphysical to have the overall rate be timestep dependent!

Reply:
fixed the `random` note.  For the 2nd note, time comes in later, in the `else` branch.  What am I missing?